src/app/filecoin-plus/allocators/components/SelectFilter.tsx

```diff
@@ -0,0 +1,78 @@
+'use client'
+
+import {
+  Listbox as HeadlessUIListbox,
+  ListboxButton as HeadlessUIListboxButton,
+  ListboxOptions as HeadlessUIListboxOptions,
+  ListboxOption as HeadlessUIListboxOption,
+} from '@headlessui/react'
+import { CaretDown, Check } from '@phosphor-icons/react/dist/ssr'
+import type { Column } from '@tanstack/react-table'
+
+import { Icon } from '@/components/Icon'
+
+import type { Allocator } from '../schemas/AllocatorSchema'
+
+type FilterOption = {
+  id: string
+  name: string
+}
+
+type SortListboxProps = {
+  column: Column<Allocator>
+  options: ReadonlyArray<FilterOption>
+  resetOptionName: string
+}
+
+const EMPTY_FILTER_VALUE = ''
+
+export function SelectFilter({
+  column,
+  options,
+  resetOptionName,
+}: SortListboxProps) {
+  const currentFilterValue =
+    column.getFilterValue()?.toString() || EMPTY_FILTER_VALUE
+
+  const resetOption: FilterOption = {
+    id: EMPTY_FILTER_VALUE,
+    name: resetOptionName,
+  }
```

Comment:
`resetOption` and `resetOptionName` throws me off a bit. What about something like..

```suggestion
    const defaultOption: FilterOption = {
      id: DEFAULT_FILTER_ID,
      name: defaultOptionLabel,
    }
```

And then

```
<SelectFilter
  column={typeColumn}
  options={typeOptions}
  defaultOptionLabel="All Types"
/>
```

Reply:
I'm happy with the renaming, good suggestions. Not a big fan of of declaring a `DEFAULT_OPTION` constant and then overwriting it with a prop.

I've reworked the component, let me know what you think of this new version :)